src/Microsoft.Azure.SignalR.Common/Utilities/ConnectionStringParser.cs

```diff
@@ -115,18 +116,27 @@ internal static ParsedConnectionString Parse(string connectionString)
                 "aad" => BuildAadAccessKey(builder.Uri, dict),
                 _ => BuildAccessKey(builder.Uri, dict),
             };
+
+            if (dict.TryGetValue(ServerEndpoint, out var serverEndpoint))
+            {
+                if (!ValidateEndpoint(serverEndpoint, out _))
+                {
+                    throw new ArgumentException($"{ServerEndpoint} property in connection string is not a valid URI: {serverEndpoint}.");
+                }
+            }
             return new ParsedConnectionString()
             {
                 Endpoint = builder.Uri,
-                ClientEndpoint = clientEndpoint,
+                ClientEndpoint = clientEndpoint == null ? null : new Uri(clientEndpoint),
```

Comment:
question: why new Uri from `clientEndpoint` instead of use `clientEndpoint` directly?